x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+import type {
+  AgentPolicyServiceInterface,
+  AgentService,
+  PackagePolicyServiceInterface,
+  PackageService,
+} from '@kbn/fleet-plugin/server';
+import type { GetAgentPoliciesResponseItem } from '@kbn/fleet-plugin/common';
+import moment, { MomentInput } from 'moment';
+import { CLOUD_SECURITY_POSTURE_PACKAGE_NAME, STATUS_ROUTE_PATH } from '../../../common/constants';
+import type { CspAppContext } from '../../plugin';
+import type { CspRouter } from '../../types';
+import type { CspSetupStatus, Status } from '../../../common/types';
+import {
+  addRunningAgentToAgentPolicy,
+  getCspAgentPolicies,
+  getCspPackagePolicies,
+} from '../../lib/fleet_util';
+import { isLatestFindingsIndexExists } from '../../lib/is_latest_findings_index_exists';
+
+export const INDEX_TIMEOUT_IN_MINUTES = 10;
+
+const getHealthyAgents = (enrichedAgentPolicies: GetAgentPoliciesResponseItem[]): number =>
+  enrichedAgentPolicies.reduce(
+    (previousValue, currentValue) => previousValue + (currentValue.agents || 0),
```

Comment:
during testing, I've noticed that it returns the total amount of agents and NOT only the healthy agents (online)
this will cause the user to get `index-timeout` although it is `not-deployed`